virtualizarr/readers/zarr.py

```diff
@@ -0,0 +1,302 @@
+from __future__ import annotations
+
+import asyncio
+from pathlib import Path  # noqa
+from typing import TYPE_CHECKING, Iterable, Mapping, Optional
+
+from xarray import Dataset, Index, Variable
+
+from virtualizarr.manifests import ChunkManifest, ManifestArray
+from virtualizarr.manifests.manifest import validate_and_normalize_path_to_uri  # noqa
+from virtualizarr.readers.common import (
+    VirtualBackend,
+    construct_virtual_dataset,
+    maybe_open_loadable_vars_and_indexes,
+)
+from virtualizarr.utils import check_for_collisions
+from virtualizarr.zarr import ZArray
+
+if TYPE_CHECKING:
+    import zarr
+    from zarr.core.common import concurrent_map
```

Comment:
A cautionary note: `zarr.core` is meant to be private API.

Reply:
So I'm guessing importing this from the Zarr private api is a big no-no?

Reply:
If the implementation of concurrent_map isn't too complicated you could potentially vendor it instead.